hail/src/main/scala/is/hail/expr/ir/functions/ArrayFunctions.scala

```diff
@@ -88,28 +88,28 @@ object ArrayFunctions extends RegistryFunctions {
   }
 
   def registerAll() {
-    registerIR("isEmpty", TArray(tv("T")), TBoolean)(isEmpty)
+    registerIR("isEmpty", TArray(tv("T")), TBoolean)((_, a) => isEmpty(a))
```

Comment:
This change happens in a lot of places.  I wouldn't change it now that it is done, but since type parameters are relatively rare, it might have been faster for you either overloaded registerIR to a have a version without type parameters, or else added something like `registerIRWithTypes`.

Reply:
can't overload due to same types after erasure, but agree with registerIRWithTypes

Reply:
I didn't go this route because: 1) we would need 2 different registries (irRegistry, irRegistryWithType), 2) last time I added a a "withTypes" it wasn't well received.